src/test/java/com/systelab/seed/allergy/AllergyResourceTest.java

```diff
@@ -8,59 +8,63 @@
+    private Allergy getAllergyData(String name, String signs, String symptoms) {
         Allergy allergy = new Allergy();
         allergy.setName(name);
         allergy.setSigns(signs);
+        if (symptoms != null)
+            allergy.setSymptoms(symptoms); //can be NULL
         return allergy;
     }
 
-    private Allergy getAllergyData(String name, String signs, String symptoms) {
-        Allergy allergy = new Allergy();
-        allergy.setName(name);
-        allergy.setSigns(signs);
-        allergy.setSymptoms(symptoms);
-        return allergy;
+    private void checkAllergyData(Allergy expected, Allergy actual){
+        TestUtil.checkObjectIsNotNull("Allergy", actual);
+        TestUtil.checkField("Name", expected.getName(), actual.getName());
+        TestUtil.checkField("Sign", expected.getSigns(), actual.getSigns());
+        if (expected.getName() != null)
+            TestUtil.checkField("Symptoms", expected.getSymptoms(), actual.getSymptoms());
     }
 
     @Description("Create a new allergy with name, sign and symptoms")
     @Test
     public void testCreateAllergy() {
         String expectedName = "Tree pollen";
         String expectedSigns = "Watering eyes";
-        Allergy allergy = getAllergyData(expectedName, expectedSigns);
-        Allergy allergyCreated = given().body(allergy)
-                .when().post("/allergies/allergy")
-                .then().assertThat().statusCode(200)
-                .extract().as(Allergy.class);
-        TestUtil.checkObjectIsNotNull("Allergy", allergyCreated);
-        TestUtil.checkField("Name", expectedName, allergyCreated.getName());
+
+        Allergy allergy = getAllergyData(expectedName, expectedSigns, null);
+        Response response = given().body(allergy)
+            .when().post("/allergies/allergy");
+        Allergy allergyCreated = response.then().extract().as(Allergy.class);
+        int status = response.then().extract().statusCode();
+        TestUtil.checkField("Status Code", 200, status);
+        checkAllergyData(allergy, allergyCreated);
     }
 
-    @Description("Create a new allergy with all the information")
+
+    @Description("Create a new allergy with all information")
     @Test
-    public void testCreateAllergyWithAllInfo() {
-        String expectedName = "Animal dander";
+    public void testCreateAllergyAllInfo() {
+        String expectedName = "Tree pollen";
         String expectedSigns = "Watering eyes";
         String expectedSymptoms = "Sneezing";
-
         Allergy allergy = getAllergyData(expectedName, expectedSigns, expectedSymptoms);
-        Allergy allergyCreated = given().body(allergy)
-                .when().post("/allergies/allergy")
-                .then().assertThat().statusCode(200)
-                .extract().as(Allergy.class);
-        TestUtil.checkObjectIsNotNull("Allergy", allergyCreated);
-        TestUtil.checkField("Name", expectedName, allergyCreated.getName());
+        Response response = given().body(allergy)
```

Comment:
Create a unique method for each endpoint that returns a Response (method public Response testCreateAllergy(Allergy) to do not duplicate the code

Reply:
Here you have some examples for the creation and modification.
 
 ```
private Response doCreateAllergy(Allergy){
    return given().body(allergy).when().post("/allergies/allergy");
  }
  
  private Reponse doModifyAllergy(Allergy, id){
    return given().body(patient)
        .when().put("/patients/" + id);
  }
```